homeassistant/components/qnap/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flow to configure qnap component."""
+import logging
+
+from qnapstats import QNAPStats
+from requests.exceptions import ConnectTimeout
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.persistent_notification import create as notify_create
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.helpers import config_validation as cv
+
+from .const import DEFAULT_PORT, DEFAULT_TIMEOUT
+from .const import DOMAIN  # pylint:disable=unused-import
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class QnapConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Qnap configuration flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize."""
+        self.is_imported = False
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        self.is_imported = True
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(
+        self, user_input=None, is_imported=False
+    ):  # pylint: disable=arguments-differ
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            host = user_input.get(CONF_HOST)
+            protocol = "https" if user_input.get(CONF_SSL, False) else "http"
+            api = QNAPStats(
+                host=f"{protocol}://{host}",
+                port=user_input.get(CONF_PORT, DEFAULT_PORT),
+                username=user_input.get(CONF_USERNAME),
+                password=user_input.get(CONF_PASSWORD),
```

Comment:
Why do you use `get()` if `username` and `password` are required?

Reply:
Exactly